add gradient to RadarChart

frybitsinc opened the pull request:
I also added gradient to RadarChart @imaNNeo

frybitsinc commented:
please review this pr @imaNNeo

imaNNeo commented on lib/src/chart/radar_chart/radar_chart_data.dart:
```diff
@@ -325,33 +348,38 @@ class RadarDataSet with EquatableMixin {
-        entryRadius: lerpDouble(a.entryRadius, b.entryRadius, t),
-      );
+  static RadarDataSet lerp(RadarDataSet a, RadarDataSet b, double t) {
+    return RadarDataSet(
```
Why did you change it from a single line return to a body return style?
I asked because now we have a change for all the lines

imaNNeo commented on lib/src/chart/radar_chart/radar_chart_painter.dart:
```diff
@@ -62,10 +62,10 @@ class RadarChartPainter extends BaseChartPainter<RadarChartData> {
     drawTicks(context, canvasWrapper, holder);
     drawTitles(context, canvasWrapper, holder);
     drawDataSets(canvasWrapper, holder);
+    drawGrids(canvasWrapper, holder);
```
Why did you change the order of this draw call?
Is there any specific reason we should know?

imaNNeo commented on lib/src/chart/radar_chart/radar_chart_painter.dart:
```diff
@@ -166,6 +166,12 @@ class RadarChartPainter extends BaseChartPainter<RadarChartData> {
 
+      /// draw radar shadow
+      if (data.elevation > 0) {
+        canvasWrapper.canvas
```
I see that you're adding a shadow / elevation here.
But the PR title is `add gradient to RadarChart`

So please open separate PRs for different featuers/improvements.

imaNNeo commented:
The render logic looks good to me. But please fix the above comments.
You also need to update the doc ([radar_chart.md](https://github.com/imaNNeo/fl_chart/blob/main/repo_files/documentations/radar_chart.md)) to add the new property.

frybitsinc commented:
> The render logic looks good to me. But please fix the above comments. You also need to update the doc ([radar_chart.md](https://github.com/imaNNeo/fl_chart/blob/main/repo_files/documentations/radar_chart.md)) to add the new property.

i just applied 4 issues you mentioned. if theres any issues please let me know, thank you!

imaNNeo commented:
> > The render logic looks good to me. But please fix the above comments. You also need to update the doc ([radar_chart.md](https://github.com/imaNNeo/fl_chart/blob/main/repo_files/documentations/radar_chart.md)) to add the new property.
> 
> i just applied 4 issues you mentioned. if theres any issues please let me know, thank you!

Perfect!
It's approved in my point of view.
The only thin is that you have to write unit-tests to be able to merge it.
Also, I suggest to rename `gradient` to `fillGradient` to be consistent with the `fillColor` property.

frybitsinc commented:
Thank you for detailed feedback!

- I renamed `gradient` to `fillGradient` 
- and also added unit-tests for `fillGradient`

Code verification failed so I thought of changing deprecated `tooltipRoundedRadius` to `tooltipBorderRadius`, 
but seems like its different issue ... so I didnt consider that.

frybitsinc commented:
> Hi, thanks for your quick fix. I also fixed the pipeline issue in the main branch. So I think you just need to rebase your branch, then we're good to go!

oh I see just rebased my branch to main, thank you!!

imaNNeo commented:
Thanks for your contribution!
